srcpkgs/brother-brscan4/template

```diff
@@ -0,0 +1,73 @@
+# Template file for 'brother-brscan4'
+pkgname=brother-brscan4
+version=0.4.4
+revision=1
+only_for_archs="i686 x86_64"
+depends="sane"
+short_desc="SANE scanner driver for brscan4-compatible printers"
+maintainer="Martijn van Buul <martijn.van.buul@gmail.com>"
+license="custom"
+homepage="http://support.brother.com/"
+repository="nonfree"
+nopie=yes
+_mylibrary="libsane-brother4.so.1.0.7"
+conf_files="/etc/opt/brother/scanner/brscan4/Brsane4.ini /etc/opt/brother/scanner/brscan4/brsanenetdevice4.cfg /etc/opt/brother/scanner/brscan4/models4/*.ini"
+
+if [ "$XBPS_TARGET_MACHINE" = "x86_64" ]; then
+	_debpkgid="4.amd64"
+	distfiles="http://download.brother.com/welcome/dlf006645/brscan4-${version}-${_debpkgid}.deb"
+	checksum=cd255c56e4b10125223315b64eb1205864ab8c759b6f8068c24a076f52bfc577
+	_deblibdir="usr/lib64"
+else
+	_debpkgid="4.i386"
+	distfiles="http://download.brother.com/welcome/dlf006646/brscan4-${version}-${_debpkgid}.deb"
+	checksum=6292feb076a97004c5a73ce8fecd7933d29309322869c1f26f77dbe4fe92b08e
+	_deblibdir="usr/lib"
+fi
+
+# append shrinkwrap license
+distfiles+=" http://www.brother.com/agreement/English_sane/agree.html>LICENSE.html"
+checksum+=" 3434bca1936d6a5fd6afd810cde7e1876dd4d1496722b09af180278480f464f2"
```

Comment:
IMHO, it would be simpler to just download the license at a moment, during the `do_extract()` phase, for example:
````
$XBPS_FETCH_CMD http://www.brother.com/agreement/English_sane/agree.html
````
Then `vlicense agree.html` in the `do_install()` phase.

Reply:
I strongly disagree. I don't see any benefits - only drawbacks:

* do_extract is the wrong phase to download files. Any custom retrieval really needs to be done in do_fetch instead; doing it during do_extract means downloading the file (and therefore: requires having an active connection) for *each and every* build.
* Overriding do_fetch implies that the default implementation will not be used. This means that the other "source" file (the .deb) also needs to be fetched manually. 
* Only running XBPS_FETCH_CMD really isn't good or secure enough, as no validity checking will be done. And yes, any sudden changes in the license should be flagged and given the same treatment as sudden changes in a source tarball. A call to XBPS_DIGEST_CMD (and comparing the result to a known value, and generating an error in case of a mismatch, and...) is required.
* All of this means re-creating a significant part of do_fetch(), with all risks and maintenance costs involved. At the same time, there's nothing that the original do_fetch() did *wrong*. do_fetch() is designed to handle multiple distfiles.
* One could avoid re-implementing do-fetch() by using the (undocumented) pre- or post-fetch hooks, but that's just as hacky (and would still entail duplicating code).

Reply:
@pienjo it's okay.